src/core/mantle/evaluator/simple/index.ts

```diff
@@ -111,11 +120,41 @@ function evalExpression(
       if (typeof styleExpression === "undefined") {
         return undefined;
       } else if (typeof styleExpression === "object" && styleExpression.conditions) {
-        return new ConditionalExpression(styleExpression, feature, layer.defines).evaluate();
+        const cacheKey = JSON.stringify([
+          styleExpression,
+          getCacheableProperties(styleExpression, feature?.properties),
```

Comment:
We don't have to instantiate Expression no more, but we became to need to parse expression every time...

Reply:
Hopefully this will be less heavy with the cache for getReferences in place.